File: 9-palindrome-number/9-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x){
        String str = String.valueOf(x);
        int l = 0;
        int r = str.length()-1;
        while(l<r){
            if(str.charAt(l)!=str.charAt(r))
               return false;
            l++;
            r--;
        }
        return true;
    }
}

File: 9-palindrome-number/9-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x){
        
        if(x<0)
            return false;
        else if(x==0)
            return true;
        else{
             long long rem , n = x ,rev = 0;
            while(n!=0){    
                rem=n%10;      
                rev=rev*10+rem;    
                n/=10;    
              }
            if(x==rev)
                return true;
            else
                return false;
        }
    }
};